### PR TITLE
Feature/qeauto 100

### PR DESCRIPTION
**Pull request by agarecajala**
Scenario for ticket QEAUTO-100 test C3194

### CODE REVIEW

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+namespace IDT.XGen.Main.UI.Pages.Retailers.DomesticBillPay
+{
+    using IDT.Core.Automation.Page.Base;
+    using IDT.Core.Automation.Selenium;
+    using IDT.XGen.Main.UI.Entities;
+    using IDT.XGen.Main.UI.Pages.Common;
+    using OpenQA.Selenium;
+
+    public class BillerForm : BasePage, IForm
```
Comment (wMercadoJala) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Add doc for the class

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs [hunk 2]
```diff
@@ -171,6 +174,51 @@ public void ClickOneButtonOfAList(string data, IList<IWebElement> rowsList, ILis
...
+        {
+            IList<IWebElement> rowsList = this.GetColumnElementsInTransaccionsTableByColumnName("LAST NAME");
+            string searchedData = string.Empty;
+            int position = 0;
+            foreach (IWebElement row in rowsList)
+            {
+                if (row.Text.ToUpper().Replace(" ", string.Empty).Equals(senderName.ToUpper().Replace(" ", string.Empty)))
+                {
+                    searchedData = itemsList[position].Text;
+					break;
```
Comment (wMercadoJala) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Why those unnecessary spaces

File: IDT.XGen/Main/UI/Entities/Retailers/Biller.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿// <copyright file="Biller.cs" company="IDT">
+// Copyright (c) IDT. All rights reserved.
+// </copyright>
+
+namespace IDT.XGen.Main.UI.Entities.Retailers
+{
+    /// <summary>
+    /// This class contains all attributes belonged to a biller.
+    /// </summary>
+    public class Biller : IEntity
```
Comment (DavidVallejosT) [inline IDT.XGen/Main/UI/Entities/Retailers/Biller.cs]: Change name for BillerEntity

File: IDT.XGen/Main/UI/Entities/Retailers/Biller.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public string Name { get; set; }
```
Comment (DavidVallejosT) [inline IDT.XGen/Main/UI/Entities/Retailers/Biller.cs]: Add XML comment for each variables.

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs [hunk 5]
```diff
@@ -127,17 +128,19 @@ public void ClickOnBtnSearchBy(string button)
...
+        public void ClickOnATransactionButton(string buttonName, string senderName)
         {
             IList<IWebElement> senderCols = this.GetColumnElementsInTransaccionsTableByColumnName("LAST NAME");
-            IList<IWebElement> changeBillerButtons = this.GetColumnElementsInTransaccionsTableByColumnName("CHANGE BILLER");
+            IList<IWebElement> changeBillerButtons = this.GetColumnElementsInTransaccionsTableByColumnName(buttonName.ToUpper());
             this.ClickOneButtonOfAList(senderName, senderCols, changeBillerButtons);
         }
 
-        /// <summary>
+
```
Comment (DavidVallejosT) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Remove blank line

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs [hunk 6]
```diff
@@ -147,8 +150,8 @@ public void ClickOnBtnBillerDetails(string billerName)
             IList<IWebElement> billerRows = this.GetRowElementsInBillerList();
             IList<IWebElement> billerDetailsButtons = this.GetColumnElementsInBillersListByColumnName("BILLER DETAILS");
             this.ClickOneButtonOfAList(billerName, billerRows, billerDetailsButtons);
-        }
-
+        }  
+
```
Comment (DavidVallejosT) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Code must not contain trailing  whitespace

File: IDT.XGen/Tests/UI/Features/Retailers/DomesticBillPay.feature [hunk 7]
```diff
@@ -38,6 +38,18 @@ Scenario: C3193 Check Change the Biller
...
 @smoke
+Scenario: C3194 Check Repeat the transaction
+	Given I login to "Retailer" portal web with "Retailer" credentials
+	When I click on Domestic Bill Pay option in Products menu
+		And I enter 9734382977 phone number to search transactions
+		And I click on the phone number search button
+		And I click on the Repeat in the transaction that has the following data:
+		| FirstName | LastName |
+		|   Jameson | Dickson  |
+		Then the data in Sender form should be displayed correctly
```
Comment (DavidVallejosT) [inline IDT.XGen/Tests/UI/Features/Retailers/DomesticBillPay.feature]: Review tabulation

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        [IDTFindBy(File = "BillerLocators", Locator = "CboPaymentType")]
+        private IWebElement CboPaymentType;
+
+        /// <inheritdoc/>
+        public void FillForm(IEntity entity)
+        {
+        }
+
+        /// <summary>
+        /// Gets the  value in LblBillerName.
```
Comment (ArielWRojas) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: please delete a space between the word "the" and "value"

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+        /// </summary>
+        /// <returns>The content text in LblBillerName.</returns>
+        public string GetLblBillerNameValue()
+        {
+            return this.LblBillerName.Text;
+        }
+
+        /// <summary>
+        /// Gets the  value in TxtAccountId.
```
Comment (ArielWRojas) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Maybe you can update the documentation for this method to:
Gets the account id value.

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+        /// </summary>
+        /// <returns>The content text in TxtAccountId.</returns>
+        public string GetTxtAccountIdValue()
+        {
+            return this.TxtAccountId.GetAttribute("value");
+        }
+
+        /// <summary>
+        /// Gets the  value in CboPaymentType.
```
Comment (ArielWRojas) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Maybe you can update the comment to: Gets the payment type value.

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs (hunk 9 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Thanks, the documentation in all similar methods in the entire project have this format,  that is the reason I used this format. And I think that it is better have just one format.

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs (hunk 10 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Thanks, the documentation in all similar methods in the entire project have this format,  that is the reason I used this format. And I think that it is better have just one format.

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs (hunk 8 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Thanks, the documentation in all similar methods in the entire project have this format, that is the reason I used this format. And I think that it is better have just one format.

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs [hunk 11]
```diff
@@ -171,6 +174,51 @@ public void ClickOneButtonOfAList(string data, IList<IWebElement> rowsList, ILis
...
+            return billerInfo;
+        }
+
+        /// <summary>
+        /// Gets the data contained in a specific column and row inside transaction table
+        /// </summary>
+        /// <param name="senderName">Is the Last Name data of a transaction.</param>
+        /// <param name="itemsList">Are the elements inside the AccountId or Biller column.</param>
+        /// <returns>Is the specific data in the row that contains the LastName data.</returns>
+        public string GetSpecificDataInTransactionsTable(string senderName, IList<IWebElement> itemsList)
```
Comment (ArielWRojas) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: if you get the data contained in a specific column

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs (hunk 1 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/BillerForm.cs]: Fixed

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs (hunk 2 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Fixed

File: IDT.XGen/Main/UI/Entities/Retailers/Biller.cs (hunk 3 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Entities/Retailers/Biller.cs]: Fixed

File: IDT.XGen/Main/UI/Entities/Retailers/Biller.cs (hunk 4 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Entities/Retailers/Biller.cs]: Fixed

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs (hunk 5 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Fixed

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs (hunk 6 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Fixed

File: IDT.XGen/Tests/UI/Features/Retailers/DomesticBillPay.feature (hunk 7 above)
Comment (agarecajala) [reply IDT.XGen/Tests/UI/Features/Retailers/DomesticBillPay.feature]: Fixed

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs [hunk 12]
```diff
@@ -171,6 +174,51 @@ public void ClickOneButtonOfAList(string data, IList<IWebElement> rowsList, ILis
...
+        {
+            IList<IWebElement> rowsList = this.GetColumnElementsInTransaccionsTableByColumnName("LAST NAME");
```
Comment (ArielWRojas) [inline IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: if you get the data contained of a specific column the "LAST NAME" value should be sent as parameter

File: IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs [hunk 13]
```diff
@@ -88,6 +91,14 @@ public void TheDataInSenderFormShouldBeDisplayedCorrectly()
             Assert.AreEqual(this.sender.Email, this.senderForm.GetTxtEmailValue());
         }
 
+        [Then(@"the data in Biller form should be displayed correctly")]
+        public void TheDataInBillerFormShouldBeDisplayedCorrectly()
+        {
+            Assert.AreEqual(this.biller.Name, this.billerForm.GetLblBillerNameValue());
```
Comment (ArielWRojas) [inline IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs]: please add a assertion message

File: IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs [hunk 14]
```diff
@@ -88,6 +91,14 @@ public void TheDataInSenderFormShouldBeDisplayedCorrectly()
...
+            Assert.AreEqual(this.biller.AccountId, this.billerForm.GetTxtAccountIdValue());
```
Comment (ArielWRojas) [inline IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs]: please add a assertion message

File: IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs [hunk 15]
```diff
@@ -88,6 +91,14 @@ public void TheDataInSenderFormShouldBeDisplayedCorrectly()
...
+            Assert.AreEqual(this.biller.PaymentType, this.billerForm.GetCboPaymentTypeValue());
```
Comment (ArielWRojas) [inline IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs]: please add a assertion message

Comment (agarecajala) [reply IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs]: Fixed

File: IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs (hunk 14 above)
Comment (agarecajala) [reply IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs]: Fixed

File: IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs (hunk 13 above)
Comment (agarecajala) [reply IDT.XGen/Tests/UI/StepDefinitions/Retailers/DomesticBillPaySteps.cs]: Fixed

File: IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs (hunk 12 above)
Comment (agarecajala) [reply IDT.XGen/Main/UI/Pages/Retailers/DomesticBillPay/SearchBillPayPage.cs]: Fixed

**Pull request merged**